api/clients/v2/payload_disperser.go

```diff
@@ -0,0 +1,222 @@
+	}
+
+	ethClient, err := geth.NewClient(ethConfig, gethcommon.Address{}, 0, logger)
+	if err != nil {
+		return nil, fmt.Errorf("new eth client: %w", err)
+	}
+
+	certVerifier, err := verification.NewCertVerifier(*ethClient, payloadDisperserConfig.EigenDACertVerifierAddr)
+	if err != nil {
+		return nil, fmt.Errorf("new cert verifier: %w", err)
+	}
+
+	return &PayloadDisperser{
+		logger:          logger,
+		config:          payloadDisperserConfig,
+		codec:           codec,
+		disperserClient: disperserClient,
+		certVerifier:    certVerifier,
+	}, nil
+}
+
+// SendPayload executes the dispersal of a payload, with these steps:
+//
+//  1. Encode payload into a blob
+//  2. Disperse the blob
+//  3. Poll the disperser with GetBlobStatus until a terminal status is reached, or until the polling timeout is reached
+//  4. Construct an EigenDACert if dispersal is successful
+//  5. Verify the constructed cert with a call to an ethereum contract
+//  6. Return the valid cert
+func (pd *PayloadDisperser) SendPayload(
+	ctx context.Context,
+	// payload is the raw data to be stored on eigenDA
+	payload []byte,
+	// salt is added while constructing the blob header
+	// This salt should be utilized if a blob dispersal fails, in order to retry dispersing the same payload under a
+	// different blob key.
+	salt uint32,
+) (*verification.EigenDACert, error) {
+
+	blobBytes, err := pd.codec.EncodeBlob(payload)
+	if err != nil {
+		return nil, fmt.Errorf("encode payload to blob: %w", err)
+	}
+	pd.logger.Debug("Payload encoded to blob")
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, pd.config.DisperseBlobTimeout)
+	defer cancel()
+	blobStatus, blobKey, err := pd.disperserClient.DisperseBlob(
+		timeoutCtx,
+		blobBytes,
+		pd.config.BlobVersion,
+		pd.config.Quorums,
+		salt)
+	if err != nil {
+		return nil, fmt.Errorf("disperse blob: %w", err)
+	}
+	pd.logger.Debug("Successful DisperseBlob", "blobStatus", blobStatus, "blobKey", blobKey)
+
+	timeoutCtx, cancel = context.WithTimeout(ctx, pd.config.BlobCertifiedTimeout)
+	defer cancel()
+	blobStatusReply, err := pd.pollBlobStatus(timeoutCtx, blobKey, blobStatus.ToProfobuf())
+	if err != nil {
+		return nil, fmt.Errorf("poll blob status: %w", err)
+	}
+	pd.logger.Debug("Blob status CERTIFIED", "blobKey", blobKey)
+
+	timeoutCtx, cancel = context.WithTimeout(ctx, pd.config.ContractCallTimeout)
+	defer cancel()
+	nonSignerStakesAndSignature, err := pd.certVerifier.GetNonSignerStakesAndSignature(
+		timeoutCtx, blobStatusReply.GetSignedBatch())
+	if err != nil {
+		return nil, fmt.Errorf("get non signer stake and signature: %w", err)
+	}
+	pd.logger.Debug("Retrieved NonSignerStakesAndSignature", "blobKey", blobKey)
+
+	eigenDACert, err := verification.BuildEigenDACert(
+		blobStatusReply.GetBlobInclusionInfo(),
+		blobStatusReply.GetSignedBatch().GetHeader(),
+		nonSignerStakesAndSignature)
+	if err != nil {
+		return nil, fmt.Errorf("build eigen da cert: %w", err)
```

Comment:
can we create a function for this

Reply:
Done [a5bc6241](https://github.com/Layr-Labs/eigenda/pull/1159/commits/a5bc6241d0b595081a5efd320a98255f10644910)